.github/workflows/security-scan.yml

```
@@ -0,0 +1,218 @@
+          npm install
+          sudo curl -SL https://github.com/docker/compose/releases/download/v2.10.2/docker-compose-linux-x86_64 -o /usr/local/bin/docker-compose
+          sudo chmod +x /usr/local/bin/docker-compose
+          docker pull ${DOCKER_REPO}:${DOCKER_APPLICATION_TAG}
+
+      - name: Run dynamic security scan
+        run: |
+          npm run scan:zap
+          cat zap-working-dir/zap-report.md >> $GITHUB_STEP_SUMMARY
+
+      - name: Upload ZAP logs
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)